pinot-common/src/main/java/org/apache/pinot/common/response/BrokerResponse.java

```diff
@@ -123,6 +127,17 @@ String toJsonString()
    */
   long getMinConsumingFreshnessTimeMs();
 
+  /**
+   * Get the max number of rows seen by a single operator in the query processing chain.
+   * <p>
+   * In single-stage this value is equal to {@link #getNumEntriesScannedPostFilter()} because single-stage operators
+   * cannot generate more rows than the number of rows received. This is not the case in multi-stage operators, where
+   * operators like join or union can emit more rows than the ones received on each of its children operators.
+   */
+  default long getMaxRowsInOperator() {
```

Comment:
I'd suggest not adding new stats on the top level.
In the future we might want to re-design the UI for V2 stats, and all the V2 stats should come from `StatMap`.
If this stats is truly useful, we may revisit it in a separate PR.

Reply:
This is not a new method but a method that existed in `BrokerResponseNative` and was inherited by `BrokerResponseV2` . Given it is already there (even it doesn't make sense) and it was used to calculate `isPartialResponse` I think it is useful to have it in the base interface now that `BrokerResponseNativeV2` does not inherit `BrokerResponseNative` anymore.

The alternative would be to do not make `isPartialResponse` `default` and include the same code in both `BrokerResponseNative` and `BrokerResponseNativeV2`. If you think that is better I can change it, but it looks a bit repetitive. For sure currently how this stats are being declared doesn't make sense, but we can try to improve that in the future

Reply:
To me this is actually a little bit confusing. Correct me if my understanding is wrong:
- In `v1` it is a query level stats on entries (rows * columns projected), which is the total `numEntriesScannedPostFilter` across all segments queried
- In `v2` it is the maximum rows returned by the same operator (there can be multiple operators of the same type, but they are counted separately) across all the workers. It is essentially the maximum of all the `EMITTED_ROWS` in the stats map.

In `v1` the closest matching might be `numDocsScanned`.
I feel we may discuss about this and add it as a separate PR

Reply:
Removed in 02ef0b23076064ae44b1d2496f948d6ed00dff36. We can open a future PR to add this if needed